catalog/video/catalog-mpegts-transformer/src/test/java/com/connexta/transformer/video/TestLocationKlvProcessor.java

```diff
@@ -0,0 +1,186 @@
+import ddf.catalog.data.impl.MetacardImpl;
+
+public class TestLocationKlvProcessor {
+
+    private String wkt;
+
+    private LocationKlvProcessor locationKlvProcessor;
+
+    private GeoBoxHandler klvHandler;
+
+    private Attribute attribute;
+
+    private Metacard metacard;
+
+    private KlvProcessor.Configuration klvConfiguration;
+
+    private Map<String, KlvHandler> handlers;
+
+    @Before
+    public void setup() {
+        wkt = "POLYGON ((0 0, 5 0, 5 5, 0 5, 0 0))";
+        locationKlvProcessor = new LocationKlvProcessor();
+        klvHandler = mock(GeoBoxHandler.class);
+        attribute = mock(Attribute.class);
+        when(attribute.getValues()).thenReturn(Collections.singletonList(wkt));
+        when(klvHandler.asAttribute()).thenReturn(Optional.of(attribute));
+        when(klvHandler.getAttributeName()).thenReturn(Constants.CORNER);
+        metacard = new MetacardImpl(BasicTypes.BASIC_METACARD);
+        klvConfiguration = new KlvProcessor.Configuration();
+        handlers = Collections.singletonMap(Constants.CORNER, klvHandler);
+    }
+
+    @Test
+    public void testProcess() {
+
+        klvConfiguration.set(KlvProcessor.Configuration.SUBSAMPLE_COUNT, 50);
+
+        locationKlvProcessor.process(handlers, metacard, klvConfiguration);
+
+        assertThat(metacard.getLocation(), is(wkt));
+
+    }
+
+    /**
+     * Test where the subsample count is missing from the configuration.
+     */
+    @Test
+    public void testMissingConfiguration() {
+
+        locationKlvProcessor.process(handlers, metacard, klvConfiguration);
+
+        assertThat(metacard.getLocation(), nullValue());
+
+    }
+
+    /**
+     * Test a subsample count that is below the mininum required by LocatationKlvProcessor.
+     */
+    @Test
+    public void testBadConfiguration() {
+
+        klvConfiguration.set(KlvProcessor.Configuration.SUBSAMPLE_COUNT,
+                LocationKlvProcessor.MIN_SUBSAMPLE_COUNT - 1);
+
+        locationKlvProcessor.process(handlers, metacard, klvConfiguration);
+
+        assertThat(metacard.getLocation(), nullValue());
+
+    }
+
+    /**
+     * This test iterates through a wide range of subsample inputs to make sure they all reduce to the
+     * subsample target and that there are no rounding issues.
+     */
+    @Test
+    public void testSubsample() {
+
+        int subsampleCount = 50;
+
+        String lat1 = "lat1";
+        String lon1 = "lon1";
+        String lat2 = "lat2";
+        String lon2 = "lon2";
+        String lat3 = "lat3";
+        String lon3 = "lon3";
+        String lat4 = "lat4";
+        String lon4 = "lon4";
+
+        int start = subsampleCount + 1;
+        int end = subsampleCount * 10;
+
+        for (int originalSize = start; originalSize < end; originalSize++) {
+
+            Map<String, List<Double>> rawData = new HashMap<>();
+
+            when(klvHandler.getRawGeoData()).thenReturn(rawData);
+            when(klvHandler.getLatitude1()).thenReturn(lat1);
```

Comment:
You should be able to do this `getLatitude`/`getLongitude` setup outside the loop.